### PR TITLE
Use a registry for storing docker frontend image

### PR DESCRIPTION
**Pull request by lyrixx**

### CODE REVIEW

File: .castor/utils.php [hunk 1]
```diff
@@ -98,15 +98,18 @@ function create_default_context(): Context
...
+        pty: 'dev' === $data['env'],
+        environment: create_default_environment(),
+    );
 }
 
 function docker_compose_run(
     string $runCommand,
     Context $c = null,
     string $service = 'builder',
-    string $user = 'app',
```
Comment (lyrixx) [inline .castor/utils.php:109]: It's not possible to exec something with something else than `app`. Anyway, we never changed this value, right?

File: infrastructure/docker/docker-compose.builder.yml [hunk 2]
```diff
@@ -5,19 +5,20 @@ volumes:
...
 services:
     builder:
+        image: "ghcr.io/jolicode/monologue/builder:${BUILDER_VERSION:-latest}"
         build:
             context: services/php
             target: builder
+            # cache_to:
+            #     - "ghcr.io/jolicode/monologue/builder:${BUILDER_VERSION:-latest}"
+            # cache_from:
+            #     - "ghcr.io/jolicode/monologue/builder:${BUILDER_VERSION:-latest}"
```
Comment (lyrixx) [inline infrastructure/docker/docker-compose.builder.yml:15]: If we want, we can share the cache on the registry. Not sure it's useful tho

File: infrastructure/docker/docker-compose.yml [hunk 3]
```diff
@@ -29,7 +41,7 @@ services:
             - "traefik.http.routers.${PROJECT_NAME}-frontend-unsecure.middlewares=redirect-to-https@file"
 
     postgres:
-        build: services/postgres
+        image: postgres:15.2
```
Comment (lyrixx) [inline infrastructure/docker/docker-compose.yml:44]: Avoid pushing an extra image to the registry

File: infrastructure/docker/services/php/entrypoint
Comment (lyrixx) [inline infrastructure/docker/services/php/entrypoint:1]: This file will update the linux user ID at **runtime**:

* if **I** publish an image, the image will contain **my** user id (usually 1000)
* Then if someone else (let's say GA) runs with another user id (let's say 1001), the permission will be wrong (host = 1001, container = 1000).

So this entry point will update the app user id in the container, to match the one on the host.

https://stackoverflow.com/a/66455612/685587

File: infrastructure/docker/docker-compose.builder.yml [hunk 4]
```diff
@@ -5,19 +5,20 @@ volumes:
 
 services:
     builder:
+        image: "ghcr.io/jolicode/monologue/builder:${BUILDER_VERSION:-latest}"
```
Comment (bastnic) [inline infrastructure/docker/docker-compose.builder.yml:8]: quand est-ce que ça utilise l'image / on peut build en local ?

Comment (lyrixx) [reply infrastructure/docker/docker-compose.builder.yml:8]: Par default, ca utilise toujours cette image:

* Si tu ne l'a pas ca va la pull (via `castor start`, `castor infra:pull` ou `castor up`)
* Si tu l'as déjà, ca l'utilise

Si tu veux builder en local, tu fais `castor build` et alors ca va utiliser ce que tu as builder localement. ton build va écraser l'image que tu as pu puller 

Si tu veux pousser sur le registry ce que tu as en local, tu fais `castor infra:push`

**Pull request merged**